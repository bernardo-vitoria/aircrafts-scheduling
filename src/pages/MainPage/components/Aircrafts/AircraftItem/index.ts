import AircraftItem from "./AircraftItem";

export { AircraftItem };