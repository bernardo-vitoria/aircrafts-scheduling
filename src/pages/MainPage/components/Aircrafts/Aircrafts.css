.aircraft-title {
    padding: 8px;
    text-align: center;
    padding-bottom: 4px;
};