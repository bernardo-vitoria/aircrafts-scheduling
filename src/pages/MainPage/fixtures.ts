const aircraft = { ident: "GABCD", type: "A320", economySeats: 186, base: "EGKK" };

const aircrafts = [
  { ident: "GABCD", type: "A320", economySeats: 186, base: "EGKK" },
];

const flight = {
  id: "AS1001",
  departuretime: 21600,
  arrivaltime: 26100,
  readable_departure: "06:00",
  readable_arrival: "07:15",
  origin: "LFSB",
  destination: "LFMN",
};

const flights = [
  {
    id: "AS1001",
    departuretime: 21600,
    arrivaltime: 26100,
    readable_departure: "06:00",
    readable_arrival: "07:15",
    origin: "LFSB",
    destination: "LFMN",
  },
  {
    id: "AS1002",
    departuretime: 27900,
    arrivaltime: 32100,
    readable_departure: "07:45",
    readable_arrival: "08:55",
    origin: "LFMN",
    destination: "LFSB",
  },
  {
    id: "AS1025",
    departuretime: 22800,
    arrivaltime: 28200,
    readable_departure: "06:20",
    readable_arrival: "07:50",
    origin: "LFSB",
    destination: "EDDH",
  },
  {
    id: "AS1026",
    departuretime: 30000,
    arrivaltime: 35100,
    readable_departure: "08:20",
    readable_arrival: "09:45",
    origin: "EDDH",
    destination: "LFSB",
  },
];

export {
    aircraft,
    aircrafts,
    flight,
    flights
};
