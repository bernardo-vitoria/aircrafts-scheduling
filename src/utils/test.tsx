// @ts-nocheck
import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { I18nextProvider } from "react-i18next";
import { QueryClient, QueryClientProvider } from "react-query";
import { MemoryRouter } from "react-router-dom"; // 4.0.0
import { queries } from "@testing-library/react";
import {
  render,
  RenderResult,
} from "@testing-library/react";
import i18n from "../config/i18nForTests";

const customRender = (ui: React.ReactElement, options = {}): RenderResult => {
  const CombinedProviders: React.FC = ({ children }) => {
    const config = {
      queries: {
        retry: false,
        cacheTime: 0,
        refetchOnWindowFocus: false,
        retryDelay: 0,
      },
    };

    const queryClient = new QueryClient({ defaultOptions: config });

    return (
        <QueryClientProvider client={queryClient}>
          <I18nextProvider i18n={i18n}>
            <MemoryRouter>{children}</MemoryRouter>
          </I18nextProvider>
        </QueryClientProvider>
    );
  };

  return render(ui, { wrapper: CombinedProviders, queries, ...options });
};

export {
  customRender as render
};