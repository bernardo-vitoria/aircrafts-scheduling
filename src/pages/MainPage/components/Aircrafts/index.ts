import Aircrafts from './Aircrafts';

export { Aircrafts };