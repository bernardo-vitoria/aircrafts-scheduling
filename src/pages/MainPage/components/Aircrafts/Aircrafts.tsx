import React from 'react';
import Stack from "@mui/material/Stack";
import { AircraftItem } from './AircraftItem';
import { Paper } from '../../../../components';
import { Aircraft } from '../../types';
import { useTranslation } from 'react-i18next';
import "./Aircrafts.css";

type Props = {
  aircrafts: Aircraft[];
};

const Aircrafts: React.FC<Props> = ({aircrafts}) => {
  const [t] = useTranslation();

    return (
      <Paper sx={{ height: "90vh" }}>
        <div className="aircraft-title">
          <b>{t("mainPage:aircrafts.aircrafts")}</b>
        </div>
        <Stack
          direction="column"
          justifyContent="space-evenly"
          alignItems="center"
          spacing={2}
        >
          {aircrafts.map((aircraft: any) => (
            <AircraftItem
              aircraft={aircraft}
              key={`aircraftItem-${aircraft.id}`}
            />
          ))}
        </Stack>
      </Paper>
    );
};

export default Aircrafts;