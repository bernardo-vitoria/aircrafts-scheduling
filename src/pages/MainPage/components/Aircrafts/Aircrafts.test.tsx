import Aircrafts from "./Aircrafts";
import { render } from "../../../../utils/test";
import { aircrafts } from "../../fixtures";

describe("Aircrafts", () => {
  it("should match snapshot", () => {
    const { container } = render(<Aircrafts aircrafts ={aircrafts} />);

    expect(container).toMatchSnapshot();
  });
});
