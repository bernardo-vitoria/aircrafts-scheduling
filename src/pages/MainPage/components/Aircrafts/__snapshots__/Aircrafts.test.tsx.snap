// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Aircrafts should match snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-r2vjbe-MuiPaper-root"
  >
    <div
      class="aircraft-title"
    >
      <b>
        Aircrafts
      </b>
    </div>
    <div
      class="css-1lgskws-MuiStack-root"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-f1vg69-MuiPaper-root"
      >
        <div
          class="card-header"
        >
          <div
            class="card-title"
          >
            Aircraft: 
            <b>
              GABCD
            </b>
          </div>
          <div
            class="card-icon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-ci5apu-MuiSvgIcon-root"
              data-testid="AirplanemodeActiveIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M22 16v-2l-8.5-5V3.5c0-.83-.67-1.5-1.5-1.5s-1.5.67-1.5 1.5V9L2 14v2l8.5-2.5V19L8 20.5V22l4-1 4 1v-1.5L13.5 19v-5.5L22 16z"
              />
            </svg>
          </div>
        </div>
        <div
          class="card-body"
        >
          <div
            class="card-body_column"
          >
            <div
              style="text-align: left;"
            >
              Ocupation: 
            </div>
          </div>
          <div
            class="card-body_column"
          >
            <div
              style="text-align: right;"
            >
              <b>
                0
                %
              </b>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
